cmd/spire-agent/cli/run/run.go

```diff
@@ -266,12 +267,28 @@ func (c *agentConfig) validate() error {
 		return fmt.Errorf("invalid value for trust_bundle_format, expected %q or %q", bundleFormatPEM, bundleFormatSPIFFE)
 	}
 
+	if c.TrustBundleUnixSocket != "" && c.TrustBundleURL == "" {
+		return fmt.Errorf("if trust_bundle_unix_socket is specified, so must be trust_bundle_url")
+	}
 	if c.TrustBundleURL != "" {
 		u, err := url.Parse(c.TrustBundleURL)
 		if err != nil {
 			return fmt.Errorf("unable to parse trust bundle URL: %w", err)
 		}
-		if u.Scheme != "https" {
+		if c.TrustBundleUnixSocket != "" {
+			if u.Scheme != "https" && u.Scheme != "http" {
```

Comment:
After putting some more thought on this, I think that we would support http only, at least initially. Use cases for https are not very clear for now.